lm_eval/api/task.py

```diff
@@ -1511,7 +1511,7 @@ def process_results(self, doc, results):
             # we expect multiple_targets to be a list.
             elif self.multiple_target:
                 gold = list(gold)
-            elif type(gold) != type(result):
+            elif type(gold) is type(result):
```

Comment:
`is not ` may be supposed actually
@baberabb

Reply:
nice catch!